main.py

```diff
@@ -20,107 +21,155 @@ class DnsInfo(TypedDict):
+    for record in res:
+        if record["type"] == "A":
+            record_id = record["id"]
+            record_name = record["name"]
+
+            endpoint = "zones/" + zone_id + "/dns_records/" + record_id
+            r = requests.put(
+                api_url + endpoint,
+                headers=headers,
+                json={
+                    "type": "A",
+                    "name": record_name,
+                    "ttl": "1",
+                    "content": ip_address,
+                },
+            )
 
             if r.status_code == 200:
-                logging.info("Successfully Changed {} in zone {}".format(
-                             record_name, zone_id))
+                logging.info(
+                    "Successfully Changed {} in zone {}".format(record_name, zone_id)
+                )
             else:
-                logging.error("HTTP: {} Failed to change {} in zone {}".format(
-                              r.status_code, record_name, zone_id))
+                logging.error(
+                    "HTTP: {} Failed to change {} in zone {}".format(
+                        r.status_code, record_name, zone_id
+                    )
+                )
                 logging.debug(r)
                 raise RuntimeError("Failed to change dns information")
 
 
 def update_ip(domain_name: str, new_ip: str):
-    '''If IP address in cloudflare does match provided IP, update all A-records
-    associated with this domain name on cloudflare'''
+    """If IP address in cloudflare does match provided IP, update all A-records
+    associated with this domain name on cloudflare"""
 
+    # NP: id is a reserved variable name in python - this works but breaks the builtin
     id = get_zone_id_by_name(domain_name)
     ip_addresses = get_a_record_ips(id)
     for address in ip_addresses:
         if address != new_ip:
             update_a_record_ip(id, new_ip)
         else:
-            logging.info('{} is up to date'.format(domain_name))
+            logging.info("{} is up to date".format(domain_name))
+
+
+class NoAuthKeyError(Exception):
+    ...
+
+
+class NoDomainListError(Exception):
+    ...
 
 
-if __name__ == '__main__':
-    logging.basicConfig(level=logging.INFO, format='%(asctime)s %(message)s')
+if __name__ == "__main__":
+    # NP: I'd configure this at the top....
+    # import logging
+    # logger = logging.getLogger(__name__)
+    # # then config on logger not logging (I'm not an expert here, there's lots of ways to log appropriately)
+    logging.basicConfig(level=logging.INFO, format="%(asctime)s %(message)s")
 
-    auth_key = os.environ.get('CLOUDFLARE_API_KEY')
-    domain_list = os.environ.get('CLOUDFLARE_URL_LIST').split(':')
+    # NP: the .get() should include a default value otherwise you'll get None
+    auth_key = os.environ.get("CLOUDFLARE_API_KEY", None)
+    domain_list_raw = os.environ.get("CLOUDFLARE_URL_LIST", None)
+    if auth_key is None:
```

Comment:
Doing this way to silence mypy and flake8

Reply:
When is it better to create custom exceptions vs using a default one. My thought is if I create a custom one I am forcing users of the library to catch custom exceptions instead of the standard ones

Reply:
Tough question. In application code I think custom exceptions are just a nice way to clean up error handling and give more immediate information that doesn't necessarily require reading a log message. In library code i have no real idea what the best practice is. I think my current conviction is that i have to lookup whatever error/exception would be thrown anyways,regardless of being builtin or not, so I like the custom model in general